docs/DeveloperGuide.md

```diff
@@ -1,34 +1,565 @@
+<br/>
+
+Given below is an example usage scenario showing how the `watchlist` command behaves at each step.
+
+**Step 1:** User launches the application for the first time. The `Workspace` of a user will be initialised to the initial workspace state, and the `activeWatchlist` will point to the first watchlist found in the `watchlistList` of the initialised `Workspace`.
+
+![Watchlist Command Initial State Diagram](images/WatchlistCommand-Initial-State.png) <br/>
+*Figure 16: Watchlist Command Initial State*
+
+<br/>
+
+**Step 2:** User executes `watchlist -n NewAnime` to create a new watchlist named "NewAnime". `Main` calls `Parser#getCommand()` and passes the command to it.
+
+**Step 3:** `Parser` extracts "watchlist" from the command, and according to the command type, it calls `WatchlistParser#parse()` and passes the command description, "-n NewAnime", to it.
+
+**Step 4:** `WatchlistParser` proceed to parse the command description, extract, and validate the inputs "-n" and "NewAnime". Once validated and no exception was thrown, it creates and initialises `WatchlistCommand` with the inputs and return this object to `Parser`, which then returns it to `Main`.
+
+**Step 5:** `WatchlistParser` is terminated and `Main` calls `WatchlistCommand#execute()` with `animeData`, `storageManager`, and `user` to create the new watchlist.
+
+![Watchlist Command After Step 5 Diagram](images/WatchlistCommand-After-Step-5.png) <br/>
+*Figure 17: Watchlist Command After Step 5*
+
+<br/>
+
+**Step 6:** `WatchlistCommand` first calls `User#getActiveWorkspace()` to initialise a `Workspace` object, `activeWorkspace`. It then calls `WatchlistCommand#createWatchlist()` to perform the operation and to call `StorageManager#saveWatchlist()` with `activeWorkspace.getName()` and `watchlistName` to save the newly created watchlist.
+
+**Step 7:** `WatchlistCommand` is terminated.
+
+![Watchlist Command Final State After Create Diagram](images/WatchlistCommand-Final-State-After-Create.png) <br/>
+*Figure 18: Watchlist Command Final State After Create*
+
+<br/>
+
+All the other options in the `watchlist` command also follows a similar execution process. 
+
+The following diagrams will continue from step 7, and it will show you how the state of the application changes as it continues to execute the select and delete option of the `watchlist` command. The list option (`-l`) is not shown as there is no change in the application state after its execution.
+
+<br/>
+
+The user executes `watchlist -s 2` to change his active watchlist to the second watchlist ("NewAnime") in the list.
+
+![Watchlist Command State After Select Diagram](images/WatchlistCommand-After-Select.png) <br/>
+*Figure 19: Watchlist Command State After Select*
+
+<br/>
+
+The user executes `watchlist -d 2` to delete the second watchlist ("NewAnime") in the list.
+
+![Watchlist Command State After Delete Diagram](images/WatchlistCommand-After-Delete.png) <br/>
```

Comment:
Words are too small in the sequence diagram. Hard for the user to read and follow